internal/pkg/manifest/svc.go

```diff
@@ -662,34 +667,46 @@ func prepareParsedExposedPortsMap(exposedPorts []ExposedPort) (map[string][]Expo
 	return parsedContainerMap, parsedExposedPortMap
 }
 
-// NetworkLoadBalancerTarget returns target container and target port for the NLB configuration.
+// Target returns target container and target port for the NLB Listener configuration.
 // This method should be called only when NLB config is not empty.
-func (s *LoadBalancedWebService) NetworkLoadBalancerTarget() (targetContainer string, targetPort string, err error) {
+func (l NetworkLoadBalancerListener) Target(exposedPorts ExposedPortsIndex) (targetContainer string, targetPort string, err error) {
 	// Configure target container and port.
-	targetContainer = aws.StringValue(s.Name)
-	if s.NLBConfig.Listener.TargetContainer != nil {
-		targetContainer = aws.StringValue(s.NLBConfig.Listener.TargetContainer)
-	}
-
+	targetContainer = exposedPorts.WorkloadName
 	// Parse listener port and protocol.
-	port, _, err := ParsePortMapping(s.NLBConfig.Listener.Port)
+	port, _, err := ParsePortMapping(l.Port)
 	if err != nil {
 		return "", "", err
 	}
 	// By default, the target port is the same as listener port.
 	targetPort = aws.StringValue(port)
-	if targetContainer != aws.StringValue(s.Name) {
-		// If the target container is a sidecar container, the target port is the exposed sidecar port.
-		sideCarPort := s.Sidecars[targetContainer].Port // We validated that a sidecar container exposes a port if it is a target container.
-		port, _, err := ParsePortMapping(sideCarPort)
-		if err != nil {
-			return "", "", err
+
+	if l.TargetContainer == nil && l.TargetPort == nil { // both targetPort and targetContainer are nil.
+		return
+	}
+
+	if l.TargetPort == nil { // when target_port is nil
+		if aws.StringValue(l.TargetContainer) != exposedPorts.WorkloadName {
+			targetContainer = aws.StringValue(l.TargetContainer)
+			for _, portConfig := range exposedPorts.PortsForContainer[targetContainer] {
+				if portConfig.Implicit {
+					targetPort = strconv.Itoa(int(portConfig.Port))
```

Comment:
We probably need a comment here.
```suggestion
					targetPort = strconv.Itoa(int(portConfig.Port)) 
					/* NOTE: When the `target_port` is empty, the intented target port should be the port that is explictly exposed by the container. Consider the following example
					```
  					http:
  					   target_container: nginx
 					   target_port: 83 # Implicitly exposed by the nginx container
					nlb:
					  port: 80/tcp
					  target_container: nginx
					 sidecars:
					   nginx:
					     port: 81 # Explicitly exposed by the nginx container.
					 ```					 
 					 In this example, the target port for the NLB listener should be 81 
					*/
					
```